### DISCUSSION TITLE
Mass-Scale Deployment of Sensors

### DISCUSSION THREAD

**Discussion by ssenthu** [openthread/openthread#9580] [Q&A] [answered]
We have several warehouses with more than 100 border routers in various locations. We have 800,000 sensors, each with an independent PSKD. What is the best way to commission them? We don't know which sensors will go to which warehouse. Each warehouse will have its own Thread network (with the same PAN ID, channel, network key, etc.) but with 5-10 border routers.

- Is it possible to load 800,000 sensor EUI64/PSKDs into the external commissioner?

- Can the external commissioner handle more than one border router from one Thread network (per warehouse)?

- Since all the routers are border routers, sensors will send the discovery request to the closest border router. Many border routers from the same network will relay the discovery request to the same external commissioner. Is that possible?

Please advise or direct me to any relevant white papers or articles.   @jwhui

**Comment by jwhui** [comment, marked as answer]
> Is it possible to load 800,000 sensor EUI64/PSKDs into the external commissioner?

It is technically possible to implement an External Commissioner that supports a large number of joining devices, but the current [ot-commissioner](https://github.com/openthread/ot-commissioner/) does not support this today.

> Can the external commissioner handle more than one border router from one Thread network (per warehouse)?

Yes, see the [ot-commissioner README](https://github.com/openthread/ot-commissioner/blob/main/src/app/cli/README.md) for more information about multi-network support.

> Since all the routers are border routers, sensors will send the discovery request to the closest border router. Many border routers from the same network will relay the discovery request to the same external commissioner. Is that possible?

Yes, it is technically possible.
